java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -99,6 +103,12 @@ public SonarComponents(FileLinesContextFactory fileLinesContextFactory, FileSyst
         registerTestCheckClasses(registrarContext.repositoryKey(), Lists.newArrayList(testCheckClasses != null ? testCheckClasses : new ArrayList<>()));
       }
     }
+    if(javaClasspath != null) {
+      javaClasspath.init();
+    }
+    if(javaTestClasspath != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)